lib/move-sync/__tests__/move-sync.test.js

```diff
@@ -0,0 +1,411 @@
+'use strict'
+
+const fs = require('graceful-fs')
+const os = require('os')
+const fse = require(process.cwd())
+const path = require('path')
+const assert = require('assert')
+const rimraf = require('rimraf')
+
+/* global afterEach, beforeEach, describe, it */
+
+function createSyncErrFn (errCode) {
+  const fn = function () {
+    const err = new Error()
+    err.code = errCode
+    throw err
+  }
+  return fn
+}
+
+const originalRenameSync = fs.renameSync
+const originalLinkSync = fs.linkSync
+
+function setUpMockFs (errCode) {
+  fs.renameSync = createSyncErrFn(errCode)
+  fs.linkSync = createSyncErrFn(errCode)
+}
+
+function tearDownMockFs () {
+  fs.renameSync = originalRenameSync
+  fs.linkSync = originalLinkSync
+}
+
+describe('moveSync()', () => {
+  let TEST_DIR
+
+  beforeEach(() => {
+    TEST_DIR = path.join(os.tmpdir(), 'fs-extra', 'move-sync')
+
+    fse.emptyDirSync(TEST_DIR)
+
+    // Create fixtures:
+    fs.writeFileSync(path.join(TEST_DIR, 'a-file'), 'sonic the hedgehog\n')
+    fs.mkdirSync(path.join(TEST_DIR, 'a-folder'))
+    fs.writeFileSync(path.join(TEST_DIR, 'a-folder/another-file'), 'tails\n')
+    fs.mkdirSync(path.join(TEST_DIR, 'a-folder/another-folder'))
+    fs.writeFileSync(path.join(TEST_DIR, 'a-folder/another-folder/file3'), 'knuckles\n')
+  })
+
+  afterEach(done => rimraf(TEST_DIR, done))
+
+  it('should not move if src and dest are the same', () => {
+    const src = `${TEST_DIR}/a-file`
+    const dest = `${TEST_DIR}/a-file`
+
+    try {
+      fse.moveSync(src, dest)
+    } catch (err) {
+      assert.ifError(err)
```

Comment:
All these `try..catch` blocks are unnecessary; if `moveSync` fails, it will automatically fail the test with the error message. Adding a `try..catch` with an `assert.ifError` is just redundant.